src/Graphics/Effect/Effect.cs

```diff
@@ -940,41 +953,71 @@ private unsafe void INTERNAL_parseEffectStruct(IntPtr effectData)
 							{
 								memSize *= mem[j].info.elements;
 							}
-							memList.Add(new EffectParameter(
+							string cachedString = String.Empty;
+							if (mem[j].info.parameter_type == MOJOSHADER_symbolType.MOJOSHADER_SYMTYPE_STRING)
```

Comment:
This block can be put below the `new EffectParameter` part, so that we don't have to have cachedString.